samwise.js

```diff
@@ -6,106 +6,100 @@
  */
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.